transport/transport.go

```diff
@@ -306,3 +320,58 @@ func NewMultiTransport(backends []http.RoundTripper,
 		Backends:        backends,
 		HandleResponses: responsesHandler}
 }
+
+// SetTransportsConfig assign transport config to Container
+func (tc *Container) SetTransportsConfig(clientConfig httphandlerConfig.Client) {
+	tc.TransportsConfig = clientConfig.Transports
+}
+
+// SelectTransport returns transport name by method, path and queryParams
+func (tc *Container) SelectTransport(method, path, queryParams string) (transportName string) {
+	_, transportName, ok := tc.TransportsConfig.GetMatchedTransport(method, path, queryParams)
+	if !ok {
+		log.DefaultLogger.Fatalf("Transport not matched with args. method: %s, path: %s, queryParams: %s", method, path, queryParams)
+	}
+	return
+}
+
+// ConfigureHTTPTransportsContainer returns RoundTrippers mapped by transport name from configuration
+func ConfigureHTTPTransportsContainer(clientConf httphandlerConfig.Client) (transportContainer Container, err error) {
+	roundTrippers := make(map[string]http.RoundTripper)
+	transportContainer.SetTransportsConfig(clientConf)
+
+	maxIdleConnsPerHost := defaultMaxIdleConnsPerHost
+	responseHeaderTimeout := defaultResponseHeaderTimeout
+	if len(clientConf.Transports) > 0 {
+		var defaultRoundTripper http.RoundTripper
+		for _, transport := range clientConf.Transports {
+			roundTrippers[transport.Name] = perepareTransport(transport.Details, maxIdleConnsPerHost, responseHeaderTimeout)
+			defaultRoundTripper = roundTrippers[transport.Name]
+		}
+		transportContainer.RoundTrippers = roundTrippers
+		transportContainer.DefaultRoundTripper = defaultRoundTripper
```

Comment:
How do we know that the chosen transport can be treated as a default on in every case? Shouldn't the default transport be the `DefaultTransport` from config file?

Reply:
Done